### PR TITLE
Cards a1

### PR DESCRIPTION
**Pull request by Zain-MKhan**

### CODE REVIEW

**Comment by Zain-MKhan** [comment]
Approve and merge when you can

**Comment by uzairali245** [comment]
im getting some errors when i compile with this `g++ src/Main.cpp src/CardsDriver.cpp src/Cards.cpp -o gameEngine.out` is it same for u/

**Review by HAOlleik — APPROVED** [review]
LGTM

**Review by Yason287 — COMMENTED** [review]
Looks good

File: src/Cards.h [hunk 1]
```diff
@@ -0,0 +1,108 @@
...
+// forward declarations
+class Player;
+
+class Card
+{
+    private:
+        Player *_player;
+
+    protected:
+        int cardID;
```
Comment (daxsis) [inline src/Cards.h:20]: in the assignment it is said to have all pointers

File: src/Cards.h [hunk 2]
```diff
@@ -0,0 +1,108 @@
...
+        string cardType;
```
Comment (daxsis) [inline src/Cards.h:21]: same

File: src/Cards.h [hunk 3]
```diff
@@ -0,0 +1,108 @@
...
+        int getCardID();
+
+        //void play(Player &p);
+        //bool validate();
+};
+
+class Deck
+{
+    protected:
+        vector <Card*> deckVector; //vector of pointers to objects of type Card
```
Comment (daxsis) [inline src/Cards.h:42]: same

**Review by daxsis — DISMISSED** [review]

**Review by uzairali245 — APPROVED** [review]
Lgtm

**Review by daxsis — APPROVED** [review]

**Pull request merged**